### DISCUSSION TITLE
Next Js 13 - loading.tsx - SEO concern

### DISCUSSION THREAD

**Discussion by dumbravaandrei22** [vercel/next.js#42988] [Help] [answered]
Hello!

In the last few days, I played around NextJs 13 trying to migrate my project from Next 12 to 13.
In my use case, I have a loading UI on each page route (the first loading.tsx from the /app directory).

What I am trying to achieve is to have loading.tsx enabled only when you navigate using the app (internal next js router). When you make the first GET request (like a search engine crawler, or the first request in the browser) I think that we should not use loading.tsx giving the SEO concern.

I ran some experiments, and it seems that the very first time you accessed the page, you will see the loading.tsx. Deactivating the browser JavaScript will result in showing the loading UI forever. 

I may be totally wrong, that's why I opened this discussion.

This relates to this: https://github.com/vercel/next.js/discussions/41745#discussion-4500612

**Comment by davidhellmann** [comment]
I think you have to generate static params and fetch on build time? 

https://beta.nextjs.org/docs/data-fetching/generating-static-params

**Reply by dumbravaandrei22** [reply]
You are talking about Static pages or ISR, but imagine that I have a 100% dynamic page (dynamic page, not [slug] in the URL).

What about non-static pages (full dynamic)?
Doc here: https://beta.nextjs.org/docs/api-reference/segment-config#dynamic 

Note: At this moment you cannot obtain a dynamic (old getServersidProps) page in build mode (only dev), there is a bug opened: https://github.com/vercel/next.js/issues/42856

**Comment by icyJoseph** [comment, marked as answer]
Interesting, I made a page that reads cookies, so that'd make it fetch on demand and I didn't really liked what I saw... I had understood that if the client is using HTTP 1.1 then the loading is skipped and the HTML is flushed? Refer to these answers:

https://github.com/vercel/next.js/discussions/41745#discussioncomment-4042599

Perhaps something bugged out between versions.

**Reply by dumbravaandrei22** [reply]
@icyJoseph Thank you! Good stuff! I will have a look.

**Reply by dumbravaandrei22** [reply]
I think that FreedKiss3 from https://github.com/vercel/next.js/discussions/41745#discussioncomment-4042599 is answering my question. Thank you @icyJoseph .
